homeassistant/components/media_player/plex.py

```diff
@@ -157,99 +253,377 @@ def update_sessions():
+        self.optional_config = optional_config
         self.plex_sessions = plex_sessions
         self.update_devices = update_devices
         self.update_sessions = update_sessions
-        self.set_device(device)
-        self._season = None
 
-    def set_device(self, device):
-        """Set the device property."""
-        self.device = device
+        # Music
+        self._media_album_artist = None
+        self._media_album_name = None
+        self._media_artist = None
+        self._media_track = None
+
+        # TV Show
+        self._media_episode = None
+        self._media_season = None
+        self._media_series_title = None
+
+        self.refresh(device, session)
+
+        # Assign custom entity ID if desired
+        if self.optional_config[CONF_USE_CUSTOM_ENTITY_IDS]:
+            prefix = ''
+            # allow for namespace prefixing when using custom entity names
+            if optional_config["entity_namespace"]:
+                prefix = optional_config["entity_namespace"] + '_'
+
+            # rename the entity id
+            if self.machine_identifier:
+                self.entity_id = "%s.%s%s" % (
+                    'media_player', prefix,
+                    self.machine_identifier.lower().replace('-', '_'))
+            else:
+                if self.name:
+                    self.entity_id = "%s.%s%s" % (
+                        'media_player', prefix,
+                        self.name.lower().replace('-', '_'))
+
+    # pylint: disable=too-many-branches, too-many-statements
+    def refresh(self, device, session):
+        """Refresh key device data."""
+        # previous data
+        # frozen detection - reset if content or position changed
+        if (self._frozen_content_id is not None and
+                self._media_content_id != self._frozen_content_id) or (
+                    self._media_position != self._last_position or
+                    self._last_position_changed is None):
+            self._last_position = self._media_position
+            self._last_position_changed = datetime.now()
+            self._frozen_fixed_state = None
+            self._frozen_content_id = None
+
+        # new data refresh
+        if session:
+            self._session = session
+        if device:
+            self._device = device
+            self._session = None
+
+        if self._device:
+            self._machine_identifier = self._convert_na_to_none(
+                self._device.machineIdentifier)
+            self._name = self._convert_na_to_none(
+                self._device.title) or DEVICE_DEFAULT_NAME
+
+        # set valid session, preferring device session
+        if self._device and self.plex_sessions.get(
+                self._device.machineIdentifier, None):
+            self._session = self._convert_na_to_none(self.plex_sessions.get(
+                self._device.machineIdentifier, None))
+
+        if self._session:
+            self._media_position = self._convert_na_to_none(
```

Comment:
The AppleTV is pushed based. You are now still updating the media position on every update.

You can see this in the logs. When you are playing some media, you should not see a state change every 10 seconds.